colpali_engine/collators/custom_collator.py

```diff
@@ -85,18 +85,34 @@ def forward_vision_idefics(self, examples):
             text_query = None
             if example["query"] is not None:
                 query = example["query"] + self.suffix
-                messages_query = [
-                    {
-                        "role": "user",
-                        "content": [
-                            {
-                                "type": "text",
-                                "text": f"Question: {query}",
-                            },
-                        ],
-                    },
-                ]
-                text_query = self.processor.apply_chat_template(messages_query, add_generation_prompt=False).strip()
+                if self.processor.__class__.__name__ == "Idefics3Processor":
+                    
+                    messages_query = [
+                        {
+                            "role": "user",
+                            "content": [
+                                {
+                                    "type": "text",
+                                    "text": f"Question: {query}",
+                                },
+                                {"type": "image"},
```

Comment:
why do you need type image here ?

Reply:
is required line 323 in `transformers/src/transformers/models/idefics3/processing_idefics3.py` , if we don't use the `image_token` (`'<image>'`) the for loop breaks